src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/CreateMergedManifestBuildModelTests.cs

```diff
@@ -0,0 +1,289 @@
+        }
+
+        [Test]
+        [Ignore("Fails due to bug https://github.com/dotnet/arcade/issues/6677")]
+        public void GivenAssetsInBlobSet()
+        {
+            AssetData dataInBlobSet = pushMetadata.GetManifestAsAsset(ImmutableList.Create(nonShippingAssetData), "thisIsALocation", "thisIsTheManifestFileName");
+            BlobArtifactModel blobArtifactModel = new BlobArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                {
+                    { "NonShipping", true.ToString().ToLower() }
+                },
+                Id = dataInBlobSet.Name
+            };
+
+            expectedBuildModel.Artifacts =
+                new ArtifactSet
+                {
+                    Blobs = new List<BlobArtifactModel> { blobArtifactModel }
+                };
+            BuildModel actualModel = pushMetadata.CreateMergedManifestBuildModel(
+                ImmutableList.Create(dataInBlobSet), manifestBuildData);
+
+            actualModel.Should().BeEquivalentTo(expectedBuildModel);
+        }
+
+        [Test]
+        [Ignore("Fails due to bug https://github.com/dotnet/arcade/issues/6677")]
+        public void GivenSomeAssetsInBlobSetAndSomeNot()
+        {
+            AssetData dataInBlobSet = pushMetadata.GetManifestAsAsset(ImmutableList.Create(nonShippingAssetData), "thisIsALocation", "thisIsTheManifestFileName");
+
+            PackageArtifactModel shippingPackageArtifact = new PackageArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", false.ToString().ToLower() },
+                    },
+                Id = shippingAssetData.Name,
+                Version = shippingAssetData.Version
+            };
+
+            BlobArtifactModel blobArtifactModel = new BlobArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                {
+                    { "NonShipping", true.ToString().ToLower() }
+                },
+                Id = dataInBlobSet.Name
+            };
+
+            expectedBuildModel.Artifacts =
+                new ArtifactSet
+                {
+                    Packages = new List<PackageArtifactModel> { shippingPackageArtifact },
+                    Blobs = new List<BlobArtifactModel> { blobArtifactModel }
+                };
+
+            BuildModel actualModel = pushMetadata.CreateMergedManifestBuildModel(
+                ImmutableList.Create(dataInBlobSet, shippingAssetData), manifestBuildData);
+
+            actualModel.Should().BeEquivalentTo(expectedBuildModel);
+        }
+
+        [Test]
+        public void GivenNonShippingAssets()
+        {
+            PackageArtifactModel packageArtifact = new PackageArtifactModel
+            {
+                Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", true.ToString().ToLower() },
```

Comment:
`true.ToString().ToLower()` is just... `"true"`

Reply:
Fair enough. :D I cleaned it up since I now know that some string weirdness there wasn't what was causing the comparison to fail. :)